layout-management-services/src/main/java/org/exoplatform/layoutmanagement/rest/SiteManagementRestService.java

```diff
@@ -138,4 +141,49 @@ public Response updateSite(@Parameter(description = "site type")
       return Response.serverError().build();
     }
   }
+
+  @Path("/permissions")
+  @PATCH
+  @RolesAllowed("administrators")
+  @Operation(summary = "Update a page access and edit permission", method = "PATCH", description = "This updates the given page access and edit permission")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Page permissions updated"),
+      @ApiResponse(responseCode = "400", description = "Invalid query input"),
+      @ApiResponse(responseCode = "404", description = "Page not found"),
+      @ApiResponse(responseCode = "401", description = "Unauthorized operation"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"), })
+  public Response updateSitePermissions(@Parameter(description = "Site type", required = true)
+                                        @QueryParam("siteType")
+                                        String siteType,
+                                        @Parameter(description = "Site name", required = true)
+                                        @QueryParam("siteName")
+                                        String siteName,
+                                        @Parameter(description = "Site new edit permission", required = true)
+                                        @QueryParam("editPermission")
+                                        String editPermission,
+                                        @Parameter(description = "Site new access permissions", required = true)
+                                        @QueryParam("accessPermissions")
+                                        String accessPermissions) {
+    try {
+      if (StringUtils.isBlank(siteName) || StringUtils.isBlank(siteType) || StringUtils.isBlank(editPermission)
+          || StringUtils.isBlank(accessPermissions)) {
```

Comment:
Couldn't we update access or edit permission alone ? why should they both be non empty ?

Reply:
changed!